Tools/Scripts/webkitpy/benchmark_runner/http_server_driver/http_server/signposts/setup.py

```diff
@@ -0,0 +1,13 @@
+from distutils.core import setup, Extension
+
+module = Extension(
+    'signposts',
+    sources=['signposts.c'],
+)
+
+setup(
```

Comment:
You are going to want to leverage `AutoInstall` here. It's going to be a bit of an adventure, because we've never built a library like this one.

The place you want to start will be `Tools/Scripts/libraries/webkitcorepy/webkitcorepy/autoinstall.py`, to understand what's going on. I'm not clear exactly the right way to do this, I can broadly see two approaches:

- Add a new `Package` and `Package.Archive` subclass for your new module in `webkitpy`. This will involve overriding the `Package.archives` class and then `Package.Archvie.download`.
- Modify `Package` and `Package.Archive` to work with a static file, and register a module object in `webkitpy`'s `__init__.py`.

I'm inclined to say #2 is the right strategy, because the the subclasses would essentially be all the work you need to do for #2 anyways.